source/localizable/tutorial/service.md

```diff
@@ -1,185 +1,146 @@
+Our app needs a single function, `createMap`,
+which makes use of `google.maps.Map` to create our map element,
+`google.maps.Geocoder` to lookup the coordinates of our location,
+and `google.maps.Marker` to pin our map based on the resolved location.
 
-```app/components/location-map.js
+```app/utils/google-maps.js
 import Ember from 'ember';
 
-export default Ember.Component.extend({
-  maps: Ember.inject.service(),
+const google = window.google;
 
-  didInsertElement() {
-    this._super(...arguments);
-    let location = this.get('location');
-    let mapElement = this.get('maps').getMapElement(location);
-    this.$('.map-container').append(mapElement);
+export default Ember.Object.extend({
+
+  init() {
+    this.set('geocoder', new google.maps.Geocoder());
+  },
+
+  createMap(element, location) {
+    let map = new google.maps.Map(element, { scrollwheel: false, zoom: 10 });
+    this.pinLocation(location, map);
+    return map;
+  },
+
+  pinLocation(location, map) {
+    this.get('geocoder').geocode({address: location}, (result, status) => {
+      if (status === google.maps.GeocoderStatus.OK) {
+        let geometry = result[0].geometry.location;
+        let position = { lat: geometry.lat(), lng: geometry.lng() };
+        map.setCenter(position);
+        new google.maps.Marker({ position, map, title: location });
+      }
+    });
   }
+
 });
 ```
 
 ### Fetching Maps With a Service
 
-At this point we should have a passing component integration test. However, no maps show up when we view our web page.
-To actually generate the maps, we'll implement the maps service.
+Now that we are able to generate a map element,
+we will implement a maps service that will keep a reference to the Map object we create,
+and attach the map to an element in our application
 
 Accessing our maps API through a [service](../../applications/services) will give us several benefits
 
-* It is injected with a [service locator](https://en.wikipedia.org/wiki/Service_locator_pattern), meaning it will abstract the maps API from the code that uses it, allowing for easier refactoring and maintenance.
+* It is injected with a [service locator](https://en.wikipedia.org/wiki/Service_locator_pattern),
+  meaning it will abstract the maps API from the code that uses it,
+  allowing for easier refactoring and maintenance.
 * It is lazy-loaded, meaning it won't be initialized until it is called the first time.
-In some cases this can reduce your app's processor load and memory consumption.
-* It is a singleton, which will allow us cache map data.
-* It follows a lifecycle, meaning we have hooks to execute cleanup code when the service stops, preventing things like memory leaks and unnecessary processing.
+  In some cases this can reduce your app's processor load and memory consumption.
+* It is a singleton,
+  which will allow us to keep map data will the user navigates around the app,
```

Comment:
It is a singleton (i.e., all references to the service point to the same object, and it remains in memory until we explicitly end the service or close our app), which will allow us to keep map data __while__ the ...